### ISSUE TITLE
Chaining `Select-String` calls mistakenly passes the first call's `.Pattern` property through

### ISSUE THREAD

**Issue by mklement0** [PowerShell/PowerShell#20063]
### Prerequisites

- [X] Write a descriptive title.
- [X] Make sure you are able to repro it on the [latest released version](https://github.com/PowerShell/PowerShell/releases)
- [X] Search the existing issues.
- [X] Refer to the [FAQ](https://github.com/PowerShell/PowerShell/blob/master/docs/FAQ.md).
- [X] Refer to [Differences between Windows PowerShell 5.1 and PowerShell](https://learn.microsoft.com/powershell/scripting/whats-new/differences-from-windows-powershell).

### Steps to reproduce

```powershell
('Hello World' | Select-String Hello | Select-String World | Select-String d).Pattern
```

### Expected behavior

```console
d
```

### Actual behavior

```console
Hello

That is, the `.Pattern` value of the _first_ `Select-String` call in the chain of calls is also reported in the subsequent calls, even though they clearly specify different `-Pattern` values.
```

### Environment data

```powershell
PowerShell 7.4.0-preview.4
```

**Comment by jhoneill** [comment]
A clue to what it is doing (though not why, or if it has any benefit. )
```
('Hello World' | Select-String Hello | add-member -NotePropertyName "what" -NotePropertyValue "is this"  -PassThru | Select-String World | Select-String d) | gm
```

**Comment by 237dmitry** [comment]
It seems that Select-String interpret input object as a list of files. Like:

```powershell
dir -force | sls bashrc    # searches  'bashrc' in all of files instead of file name
```

![sls](https://github.com/PowerShell/PowerShell/assets/78153320/2c92cb5a-19d3-4ccc-84a9-5b787d530d8a)

**Comment by mklement0** [comment]
@237dmitry, yes, piping `FileInfo` instances to `Select-String` makes it search the _content_ of these files (just like passing file paths to `-LiteralPath` and `-Path` does) - this is by design, and unrelated to this bug report.

**Comment by 237dmitry** [comment]
> and unrelated to this bug report

Yes, first I wrote, then I thought
